### ISSUE TITLE
LicenseCompareHelper: GPL-2.0 license text not recognised

### ISSUE THREAD

**Issue by sdheh** [spdx/Spdx-Java-Library#217]
License text: https://www.gnu.org/licenses/old-licenses/gpl-2.0.txt
Emoji represent whether this result is expected or not.

Results:
matchingStandardLicenseIdsWithinText -> empty list ❌
matchingStandardLicenseIds -> empty list ❌
isTextStandardLicense with GPL-2.0, GPL-2.0-only or GPL-2.0-or-later -> Additional text found after the end of the expected license text starting at line #280 column #21 "end" ❌

Results when excluding optional end (keep lines 1-278):
matchingStandardLicenseIdsWithinText -> empty list ❌
matchingStandardLicenseIds -> GPL-2.0, GPL-2.0-only or GPL-2.0-or-later ✅
isTextStandardLicense with GPL-2.0, GPL-2.0-only or GPL-2.0-or-later -> No difference found ✅

I also checked the difference between the optional text in the end of the GPL-2.0 license xml template and the license text using a diff tool and didn't find any difference.

Note: tested using version 1.1.8

**Comment by goneall** [comment]
I suspect this is also related to the pattern `.{0,36000}` failing as documented in #214  Once we have a fix for that, I'll test to see if this is resolved.

**Comment by goneall** [comment]
Fixed with PR #221

**Comment by sdheh** [comment]
I just tried with version 1.1.11 and it does not work:
```
System.out.println(Arrays.toString(LicenseCompareHelper.matchingStandardLicenseIds(licenseText)));
System.out.println(LicenseCompareHelper.matchingStandardLicenseIdsWithinText(licenseText));
```
Returns
```
[]
[]
```
